### PR TITLE
Make Value own the GValue struct

### PR DESCRIPTION
**Pull request by gkoz**

### CODE REVIEW

File: glib-sys/src/lib.rs [hunk 1]
```diff
@@ -17,11 +17,11 @@
 
 extern crate libc;
 
-use libc::{c_void, c_int, c_uint, c_float, c_double, c_char, c_uchar, c_long, c_ulong};
+use libc::{c_void, c_int, c_uint, c_float, c_double, c_char, c_uchar, c_long, c_ulong, size_t};
 
 pub type GQuark = u32;
 
-pub type GType = c_ulong;
+pub type GType = size_t;
```
Comment (GuillaumeGomez) [inline glib-sys/src/lib.rs]: Where did you find this information ?

File: glib-sys/src/lib.rs [hunk 2]
```diff
@@ -34,6 +34,12 @@ pub type GCallback = extern "C" fn();
 pub type GClosureNotify = extern "C" fn(data: gpointer, closure: gpointer);
 
 #[repr(C)]
+pub struct C_GValue {
+    typ: GType,
```
Comment (GuillaumeGomez) [inline glib-sys/src/lib.rs]: typ ?

File: glib-sys/src/lib.rs (hunk 1 above)
Comment (gkoz) [inline glib-sys/src/lib.rs]: This is my current understanding of
```C
#if     GLIB_SIZEOF_SIZE_T != GLIB_SIZEOF_LONG || !defined __cplusplus
typedef gsize                           GType;
#else   /* for historic reasons, C++ links against gulong GTypes */
typedef gulong                          GType;
#endif
```
This change is a noop for unix-like architectures which are LP64 but affects Windows x64 where `sizeof(long) == 4`, so we might want to make extra sure this works there before merging. ;)

File: glib-sys/src/lib.rs (hunk 2 above)
Comment (gkoz) [inline glib-sys/src/lib.rs]: Can make this `type_` I guess. Prefixing underscores has a meaning in Rust so I'm against that.

File: glib-sys/src/lib.rs (hunk 1 above)
Comment (gkoz) [inline glib-sys/src/lib.rs]: And `GType` indeed is 8 bytes on win64 with both GTK SDKs I have, so `ulong` would've been a problem there.

**Comment by gkoz** [comment]
Fixed the field name.

**Comment by GuillaumeGomez** [comment]
I'll merge it but I'd prefer `type_` than `typ`. Can you change that please ?

**Comment by gkoz** [comment]
Yeah, it's done already.

**Comment by GuillaumeGomez** [comment]
Perfect ! Thanks for your work !

**Pull request merged**